From 79bc169e0c90c74d21d21d3d73730001a72a2dcc Mon Sep 17 00:00:00 2001
From: Jannis Pohlmann <jannis@xfce.org>
Date: Fri, 18 Sep 2009 21:41:06 +0000
Subject: Use g_utf8_strlen to calculate the sidepane width. Bugs #5390 and #3965.

Using strlen() for the number of characters is unrealistic when dealing
with multibyte strings.
---
diff --git a/thunar/thunar-shortcuts-view.c b/thunar/thunar-shortcuts-view.c
index 2ba4b61..c7df182 100644
--- a/thunar/thunar-shortcuts-view.c
+++ b/thunar/thunar-shortcuts-view.c
@@ -274,7 +274,7 @@ thunar_shortcuts_view_init (ThunarShortcutsView *view)
   renderer = g_object_new (GTK_TYPE_CELL_RENDERER_TEXT,
                            "ellipsize-set", TRUE,
                            "ellipsize", PANGO_ELLIPSIZE_END,
-                           "width-chars", strlen (_("File System")),
+                           "width-chars", g_utf8_strlen (_("File System"), -1),
                            NULL);
   g_signal_connect (G_OBJECT (renderer), "edited", G_CALLBACK (thunar_shortcuts_view_renamed), view);
   gtk_tree_view_column_pack_start (column, renderer, TRUE);
--
cgit v0.8.2.1
